### PR TITLE
feat(err): store frames

### PR DESCRIPTION
**Pull request by oliverb123**

### CODE REVIEW

File: posthog/models/error_tracking/error_tracking.py [hunk 1]
```diff
@@ -58,3 +58,50 @@ def convert_fingerprints_to_tuples(fps: list[list[str]]):
...
+    storage_ptr = models.TextField(null=True, blank=False)
+
+    class Meta:
+        indexes = [
+            models.Index(fields=["team_id", "id"]),
+        ]
+
+        constraints = [
+            models.UniqueConstraint(fields=["team_id", "id"], name="unique_id_per_team"),
+        ]
```
Comment (daibhin) [inline posthog/models/error_tracking/error_tracking.py:87]: We want these to be unique on `ref` not `id`, right?

File: posthog/models/error_tracking/error_tracking.py [hunk 2]
```diff
@@ -58,3 +58,50 @@ def convert_fingerprints_to_tuples(fps: list[list[str]]):
...
+    ref = models.TextField(null=False, blank=False)
+    team = models.ForeignKey("Team", on_delete=models.CASCADE)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    # How we stored this symbol set, and where to look for it
+    # These are null if we failed to find a symbol set for a given reference. We store a
+    # row anyway, so if someone comes along later and uploads a symbol set for this reference,
+    # we can know which frame resolution results below to drop.
+    storage_method = models.TextField(choices=StorageMethod.choices, default=StorageMethod.S3, null=True)
+    storage_ptr = models.TextField(null=True, blank=False)
```
Comment (daibhin) [inline posthog/models/error_tracking/error_tracking.py:78]: `storage_path` is the only other name I've seen us use here

Comment (daibhin) [reply posthog/models/error_tracking/error_tracking.py:78]: My vote in order to keep this simpler is to go with `object_path` for now and assume it always points to S3. If we find out in future we need to support other methods we could add the `storage_method` column at that point

File: posthog/models/error_tracking/error_tracking.py [hunk 3]
```diff
@@ -58,3 +58,50 @@ def convert_fingerprints_to_tuples(fps: list[list[str]]):
         # converting back to list of lists before saving
         self.merged_fingerprints = [list(f) for f in merged_fingerprints]
         self.save()
+
+
+class ErrorTrackingSymbolSet(UUIDModel):
```
Comment (daibhin) [inline posthog/models/error_tracking/error_tracking.py:63]: Why did these need to be UUID models again? I forget

Comment (oliverb123) [reply posthog/models/error_tracking/error_tracking.py:63]: Multiple teams can have the same symbol set, if they e.g. pull from a shared library or something.

File: posthog/models/error_tracking/error_tracking.py (hunk 2 above)
Comment (oliverb123) [reply posthog/models/error_tracking/error_tracking.py:78]: I'm fine with simple here, sure

File: posthog/models/error_tracking/error_tracking.py (hunk 1 above)
Comment (oliverb123) [reply posthog/models/error_tracking/error_tracking.py:87]: Yes, correct

File: rust/cymbal/src/langs/js.rs [hunk 4]
```diff
@@ -81,43 +101,6 @@ impl RawJSFrame {
...
-            InvalidSourceMap(e) => Err(JsResolveErr::from(e)),
-            TokenNotFound(s, l, c) => Err(TokenNotFound(s, l, c)),
-            InvalidSourceUrl(u) => Err(InvalidSourceUrl(u)),
-            InvalidSourceMapHeader(u) => Err(InvalidSourceMapHeader(u)),
-            InvalidSourceMapUrl(u) => Err(InvalidSourceMapUrl(u)),
-        }?)
-    }
-
-    // Returns none if the frame is
-    fn try_assume_unminified(&self) -> Option<Frame> {
```
Comment (daibhin) [inline rust/cymbal/src/langs/js.rs:107]: Will removing this mean that we have no way of knowing if something is unminified i.e. does not need demangling.  Not necessary right now as long as we know to come back to it

File: rust/cymbal/src/symbol_store/caching.rs [hunk 5]
```diff
@@ -151,6 +134,27 @@ impl CacheInner {
...
+        self.len()
+    }
+}
+
+impl Cacheable for SourceMap {
+    fn bytes(&self) -> usize {
+        // This is an extremely expensive way to get the size of a sourcemap, but we're more-or-less ok with that,
+        // since we only call it when we add an item to the cache, which should be rare.
+        let mut data = vec![];
+        self.to_writer(&mut data).unwrap();
```
Comment (daibhin) [inline rust/cymbal/src/symbol_store/caching.rs:157]: Assuming it's ok to add the `unwrap` here?

**Review by daibhin — APPROVED** [review]

File: rust/cymbal/src/langs/js.rs (hunk 4 above)
Comment (oliverb123) [reply rust/cymbal/src/langs/js.rs:107]: I couldn't figure out a way to differentiate between unminified frames and simply errored ones - we should add this back once we figure out a good heuristic for that, but I think we can do that in time.

File: rust/cymbal/src/symbol_store/caching.rs (hunk 5 above)
Comment (oliverb123) [reply rust/cymbal/src/symbol_store/caching.rs:157]: Yeah it's fine, it only fails if the underlying writer fails I think, and a vec can never fail a write (or, a vec failing a write is an oom, not an `Err`).

**Pull request merged**